From fe89c906e5a8d6d5e9140226866d315e97d3bee3 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Sun, 06 May 2012 02:49:54 +0000
Subject: daemon: don't freak out if greeter dies after login finishes

If we've already logged a user in and the greeter is shutting
down and crashes, we should just ignore the crash and continue
with the login process.

Furthermore, if we explicitly kill the greeter and it dies with
a TERM signal, then we definitely don't want to treat that as
if it was a crash.
---
diff --git a/daemon/gdm-simple-slave.c b/daemon/gdm-simple-slave.c
index 7c709a6..bf48246 100644
--- a/daemon/gdm-simple-slave.c
+++ b/daemon/gdm-simple-slave.c
@@ -1032,7 +1032,9 @@ on_greeter_session_died (GdmGreeterSession    *greeter,
                          GdmSimpleSlave       *slave)
 {
         g_debug ("GdmSimpleSlave: Greeter died: %d", signal);
-        gdm_slave_stopped (GDM_SLAVE (slave));
+        if (slave->priv->start_session_service_name == NULL) {
+                gdm_slave_stopped (GDM_SLAVE (slave));
+        }
 }
 
 static void
--
cgit v0.9.0.2
